packages/ui/lib/components/Form/AddressInput.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  InputHTMLAttributes,
+  ForwardedRef,
+  ReactNode,
+  useState,
+  useEffect
+} from 'react'
+import type { Size, SizeStyleProp } from '../../types'
+import { forwardRef } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { FieldLayout } from './FieldLayout'
+import { Icon } from '../Icon/Icon'
+import { FaWallet as WalletIcon } from 'react-icons/fa'
+import { IconBaseProps } from 'react-icons'
+import { minifyAddress } from '../../utils'
+
+export interface Props
+  extends Omit<
+    InputHTMLAttributes<HTMLInputElement>,
+    'size' | 'id' | 'children'
+  > {
+  label?: string
+  size?: Size
+  description?: ReactNode
+  withIcon?: boolean
+  isTruncated?: Boolean
+  web3Service?: any
+  localForm: any;
+  name: any,
+}
+
+const SIZE_STYLES: SizeStyleProp = {
+  small: 'pl-2.5 py-1.5 text-sm',
+  medium: 'pl-4 py-2 text-base',
+  large: 'pl-4 py-2.5',
+}
+
+const STATE_STYLES = {
+  error:
+    'border-brand-secondary hover:border-brand-secondary focus:border-brand-secondary focus:ring-brand-secondary',
+  success:
+    'border-green-500 hover:border-green-500 focus:border-green-500 focus:ring-green-500',
+}
+
+const CustomizedIcon = (props: IconBaseProps) => <WalletIcon {...props} className="fill-gray-500" />
+
+/**
+ * Primary Input component for React Hook Form
+ *
+ * @param label - Label for the input
+ * @param name - Name of the input
+ * @param type - Type of the input
+ * @param localForm - React Hook Form object
+ * @returns Input component
+ *
+ */
+ export const AddressInput = forwardRef(
+  (props: Props, ref: ForwardedRef<HTMLInputElement>) => {
+    const {
+      size = 'medium',
+      value,
+      className,
+      description,
+      label,
+      withIcon = true,
+      isTruncated,
+      web3Service,
+      localForm,
+      name,
+      ...inputProps
+    } = props
+
+    if (!localForm) return null;
+    const { register } = localForm
```

Comment:
You can use useFormContext() and wrap the external component inside provider instead of passing the methods around.

Reply:
I don't think we can do it since `AddressInput` lives inside the UI which is a separate package, that's not aware of form context. I did it try it before and `useFormContext()` always returns `null`. Also, I don't see why passing the methods around is bad. People at RaidGuild also follow this approach where they pass the whole useForm hook from the parent form component. You can check [this](https://github.com/raid-guild/design-system/blob/main/src/components/forms/Input/Input.tsx#L42) out. @searchableguy

Reply:
It was a nit :D 

And you need to wrap the parent component in form context provider to use the useFormHook. 

Either way is fine.

Reply:
Sorry, I wasn't calling you anything. It is a nit as in It is not consequential and either way is fine.

Reply:
Bummer, I totally misunderstood. I'm sorry, I thought you referred lit to the person at RaidGuild who followed that particular approach.